support-frontend/assets/components/subscriptionCheckouts/address/addressFields.tsx

```diff
@@ -184,10 +189,47 @@ export function AddressFields({ scope, ...props }: PropTypes): JSX.Element {
 				data-qm-masking="blocklist"
 				label="Country"
 				value={props.country}
-				onChange={(e) => {
-					const isoCountry = Country.fromString(e.target.value);
-					if (isoCountry) {
-						props.setCountry(isoCountry);
+				onChange={(event) => {
+					const selectedCountry = Country.fromString(event.target.value);
+
+					if (selectedCountry && countryGroupId) {
+						const selectedCountryGroup = Object.entries(countryGroups).find(
+							([, countryGroup]) =>
+								countryGroup.countries.includes(selectedCountry),
+						)?.[0];
+
+						if (countryGroupId !== selectedCountryGroup) {
+							const pathname = window.location.pathname;
+							const currentInternationalisationId = pathname.split('/')[1];
+							const selectedInternationalisationId =
+								countryGroups[selectedCountryGroup as CountryGroupId]
+									.supportInternationalisationId;
+							const redirectPathname = pathname.replace(
+								currentInternationalisationId,
+								selectedInternationalisationId,
+							);
+
+							const urlSearchParams = new URLSearchParams(
+								window.location.search,
+							);
+							const product = urlSearchParams.get('product') as ProductKey;
+							const ratePlan = urlSearchParams.get('ratePlan') as string;
+							const { productKey, ratePlanKey } =
+								internationaliseProductAndRatePlan(
+									selectedInternationalisationId,
+									product,
+									ratePlan,
+								);
+							urlSearchParams.set('product', productKey);
+							urlSearchParams.set('ratePlan', ratePlanKey);
+
+							const location = `${redirectPathname}?${urlSearchParams.toString()}`;
```

Comment:
Just realised they'll lose the promoCode applied via the current URL when we redirect, hmm need to have a think about that and the best way to handle this.

Reply:
Yea, not 100% sure - currently, for `TierThree` it works as there are default promotions.

We don't have "Groups" of promo codes, although if we gave them all the same name, but only made them valid in certain groups, I think that may work too.

e.g.
- `TIER_THREE_MONTHLY` valid in `uk`
- `TIER_THREE_MONTHLY` valid in `us`
- `TIER_THREE_MONTHLY` valid in `au`
- ...

And then set `TIER_THREE_MONTHLY` as the default promo code - I think this would work.

Reply:
Nice ok, given the checkout already picks the default promo if it's available regardless of whether the `promoCode` query string is in the URL I don't think this is an issue